bgpd/bgp_route.c

```diff
@@ -254,8 +254,9 @@ static void bgp_path_info_free(struct bgp_path_info *path)
 	bgp_unlink_nexthop(path);
 	bgp_path_info_extra_free(&path->extra);
 	bgp_path_info_mpath_free(&path->mpath);
-	bgp_addpath_free_info_data(&path->tx_addpath,
-				 path->net ? &path->net->tx_addpath : NULL);
+	if (path->net)
```

Comment:
If this function is ever called elsewhere we can have the same situation where we would need to remember to not call with a if(path->net) as well.

Secondly -> I realize this is a deletion event, but have we gotten into a state where something else has gone wrong?  Might it be better to find and fix that state?

Reply:
Yes, this patch doesn't fix the root cause of the problem (`path->net` being set to NULL), it's only a bandaid to prevent a fairly common crash that happens when I test bgpd with the CLI fuzzer.

I mentioned yesterday on the #bgpd channel about a similar crash that happens in the `vpn_leak_to_vrf_update_onevrf()` function when `path->net` is set to NULL.

A proper fix would certainly be the ideal thing to do, but while we don't have that, I think it's better to handle the inconsistency rather than crashing the daemon (especially for the 7.0 release). What do you think?

Reply:
@donaldsharp opened #3846 as per your request.